### DISCUSSION TITLE
Problem In Date When change Lang from english to Arabic ?

### DISCUSSION THREAD

**Discussion by ra-bedo11** [serenity-is/Serenity#6320] [Help]
this when choose arabic language from right to left
![Error Serenity 1](https://user-images.githubusercontent.com/29665710/158040177-150e3234-c36b-4469-bcbd-a6918a09d558.jpg)

this when choose lang english left to right
![Error Serenity 2](https://user-images.githubusercontent.com/29665710/158040215-51a701e9-e516-4a75-942e-bbc4a629c927.jpg)

![Error Serenity 3](https://user-images.githubusercontent.com/29665710/158040265-fa45daa9-21ac-4f55-b546-a18af6fb537e.jpg)

this my code in userCultuerprovider

        private static Dictionary<string, string> TwoLetterToFourLetter = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
        {
            { "ar", "ar-EG" },
            { "en", "en-US" },
            { "zh", "zh-CN" },
            { "vi", "vi-VN" },
            { "fa", "fa-IR" }
        };

        public Task<ProviderCultureResult> DetermineProviderCultureResult(HttpContext httpContext)
        {
            var culture = httpContext.Request.Cookies["LanguagePreference"];
            if (string.IsNullOrEmpty(culture))
                culture = null;
            else
            {
                if (culture.Length == 2)
                {
                    string code;
                    if (TwoLetterToFourLetter.TryGetValue(culture, out code))
                        culture = code;
                    else
                        culture = culture + "-" + culture.ToUpperInvariant();
                }
            }

            return Task.FromResult(new ProviderCultureResult(culture ?? "en-US", culture ?? "en-US"));
        }

        private static List<CultureInfo> supportedCultures;
        private static readonly string[] supportedCultureIdentifiers = new string[] {
            "ar-EG",
            "de-DE",
            "en-US",
            "en-GB",
            "es-ES",
            "fa-IR",
            "it-IT",
            "pt-PT",
            "pt-BR",
            "ru-RU",
            "tr-TR",
            "vi-VN",
            "zh-CN"
        };

        public static IList<CultureInfo> SupportedCultures
        {
            get
            {
                if (supportedCultures == null)
                    supportedCultures = supportedCultureIdentifiers.Select(x =>
                    {
                        try
                        {
                            return new CultureInfo(x);
                        }
                        catch
                        {
                            return null;
                        }
                    }).Where(x => x != null).ToList();

                return supportedCultures;
            }
        }

**Comment by mohesa78** [comment]
same issue with me with last serenity core 5.1

**Reply by ra-bedo11** [reply]
I know !!!!
But what is the solution?

**Reply by mohesa78** [reply]
you can fix the format for all languages at  _LayoutHead.cshtml

<script type="application/json" id="ScriptCulture">@Html.Raw(Serenity.JSON.Stringify(new ScriptCulture{ DateFormat = "yyyy-MM-dd", DecimalSeparator = ".", DateSeparator = "-" }))</script>

**Reply by ra-bedo11** [reply]
> you can fix the format for all languages at _LayoutHead.cshtml
> 
> <script type="application/json" id="ScriptCulture">@Html.Raw(Serenity.JSON.Stringify(new ScriptCulture{ DateFormat = "yyyy-MM-dd", DecimalSeparator = ".", DateSeparator = "-" }))</script>

![Ganty_error_Languge](https://user-images.githubusercontent.com/29665710/164078774-88eddfbf-0621-47fb-99d3-fe783de1313a.png)

this code
its not work

**Comment by ra-bedo11** [comment]
lang Farsi Is Work 
But Arabic Its not working Why ?
